js/canvas.js

```diff
@@ -0,0 +1,367 @@
+
+    if (draggingAndStickingRect) {
+      const nearestStandingAndStickingRect = this.getNearestStickedRectangleToDraggingRectangle(
+        draggingAndStickingRect,
+        arrayOfStandingAndStickingRects
+      );
+
+      this.setStickedSideToDraggingRectangle(draggingAndStickingRect, nearestStandingAndStickingRect);
+
+      switch (draggingAndStickingRect.stickedSide) {
+        case 'top-left':
+          draggingAndStickingRect.stickedX = nearestStandingAndStickingRect.x;
+          draggingAndStickingRect.stickedY = nearestStandingAndStickingRect.y - draggingAndStickingRect.height;
+          break;
+        case 'top-right':
+          draggingAndStickingRect.stickedX =
+            nearestStandingAndStickingRect.x + nearestStandingAndStickingRect.width - draggingAndStickingRect.width;
+          draggingAndStickingRect.stickedY = nearestStandingAndStickingRect.y - draggingAndStickingRect.height;
+          break;
+        case 'bottom-right':
+          draggingAndStickingRect.stickedX =
+            nearestStandingAndStickingRect.x + nearestStandingAndStickingRect.width - draggingAndStickingRect.width;
+          draggingAndStickingRect.stickedY = nearestStandingAndStickingRect.y + nearestStandingAndStickingRect.height;
+          break;
+        case 'bottom-left':
+          draggingAndStickingRect.stickedX = nearestStandingAndStickingRect.x;
+          draggingAndStickingRect.stickedY = nearestStandingAndStickingRect.y + nearestStandingAndStickingRect.height;
+          break;
+        case 'left-bottom':
+          draggingAndStickingRect.stickedX = nearestStandingAndStickingRect.x - draggingAndStickingRect.width;
+          draggingAndStickingRect.stickedY =
+            nearestStandingAndStickingRect.y + nearestStandingAndStickingRect.height - draggingAndStickingRect.height;
+          break;
+        case 'left-top':
+          draggingAndStickingRect.stickedX = nearestStandingAndStickingRect.x - draggingAndStickingRect.width;
+          draggingAndStickingRect.stickedY = nearestStandingAndStickingRect.y;
+          break;
+        case 'right-top':
+          draggingAndStickingRect.stickedX = nearestStandingAndStickingRect.x + nearestStandingAndStickingRect.width;
+          draggingAndStickingRect.stickedY = nearestStandingAndStickingRect.y;
+          break;
+        case 'right-bottom':
+          draggingAndStickingRect.stickedX = nearestStandingAndStickingRect.x + nearestStandingAndStickingRect.width;
+          draggingAndStickingRect.stickedY =
+            nearestStandingAndStickingRect.y + nearestStandingAndStickingRect.height - draggingAndStickingRect.height;
+          break;
+
+        default:
+          break;
+      }
+
+      const { stickedX, stickedY } = draggingAndStickingRect;
+
+      draggingAndStickingRect.arrayOfPrevStickedCoordinates.push({ stickedX, stickedY });
+    }
+
+    if (arrayOfStandingAndStickingRects.length) {
```

Comment:
Do not need `if` statement here... if `arrayOfStandingAndStickingRects` is empty code `arrayOfStandingAndStickingRects.forEarch` will not executed